src/Checkpointer.jl

```diff
@@ -78,18 +96,188 @@ end
 This is a callback function that checkpoints all simulations defined in the current coupled simulation.
 """
 function checkpoint_sims(cs::Interfacer.CoupledSimulation)
+    t = Dates.datetime2epochms(cs.dates.date[1])
+    t0 = Dates.datetime2epochms(cs.dates.date0[1])
     for sim in cs.model_sims
         if Checkpointer.get_model_prog_state(sim) !== nothing
-            t = Dates.datetime2epochms(cs.dates.date[1])
-            t0 = Dates.datetime2epochms(cs.dates.date0[1])
             Checkpointer.checkpoint_model_state(
                 sim,
                 cs.comms_ctx,
                 Int((t - t0) / 1e3),
                 output_dir = cs.dirs.checkpoints,
             )
+            Checkpointer.checkpoint_model_cache(
+                sim,
+                cs.comms_ctx,
+                Int((t - t0) / 1e3),
+                output_dir = cs.dirs.checkpoints,
+            )
+        end
+    end
+
+    # Checkpoint the Coupler fields
+    output_dir = cs.dirs.checkpoints
+    comms_ctx = cs.comms_ctx
+    time = Int((t - t0) / 1e3)
```

Comment:
Could you move these above the for loop and use them uniformly for the calls to `checkpoint_model_state`, `checkpoint_model_cache`, and the coupler fields save?